indico/htdocs/js/indico/jquery/permissions_widget.js

```diff
@@ -0,0 +1,336 @@
+            return [
+                $('<span>', {class: 'entry-icon '  + iconClass}),
+                $('<span>', {class: 'text-normal entry-label', text: text})
+            ];
+        },
+        _renderPermissions: function(principal, permissions) {
+            var self = this;
+            var $permissions = $('<div>', {class: 'permissions-box flexrow f-a-center f-self-stretch'});
+            var $permissionsList = $('<ul>').appendTo($permissions);
+            permissions.forEach(function(item) {
+                $permissionsList.append(
+                    $('<li>', {class: 'i-label bold ' + self.options.permissionsInfo[item].css_class})
+                        .append(self.options.permissionsInfo[item].title)
+                );
+            });
+            if (principal._type !== 'DefaultEntry') {
+                $permissions.append(this._renderPermissionsButtons(principal, permissions));
+            }
+            return $permissions;
+        },
+        _renderPermissionsButtons: function(principal, permissions) {
+            var $buttonsGroup = $('<div>', {class: 'group flexrow'});
+            $buttonsGroup.append(this._renderEditBtn(principal, permissions), this._renderDeleteBtn(principal));
+            return $buttonsGroup;
+        },
+        _renderEditBtn: function(principal, permissions) {
+            if (principal._type === 'IPNetworkGroup') {
+                return $('<button>', {
+                    type: 'button',
+                    class: 'i-button text-color borderless icon-only icon-edit disabled',
+                    title: $T.gettext('IP networks cannot have management permissions')
+                });
+            } else {
+                return $('<button>', {
+                    'type': 'button',
+                    'class': 'i-button text-color borderless icon-only icon-edit',
+                    'data-href': build_url(Indico.Urls.EventPermissions, {confId: this.options.eventId}),
+                    'data-title': $T.gettext('Assign Permissions'),
+                    'data-method': 'POST',
+                    'data-ajax-dialog': '',
+                    'data-params': JSON.stringify({principal: JSON.stringify(principal), permissions: permissions})
+                });
+            }
+        },
+        _renderDeleteBtn: function(principal) {
+            var self = this;
+            return $('<button>', {
+                'type': 'button',
+                'class': 'i-button text-color borderless icon-only icon-remove',
+                'data-principal': JSON.stringify(principal)
+            }).on('click', function() {
+                var $this = $(this);
+                var title = $T.gettext("Delete entry '{0}'".format(principal.name || principal.id));
+                var message = $T.gettext("Are you sure you want to permanently delete this entry?");
+                confirmPrompt(message, title).then(function() {
```

Comment:
What we *do* need to restore here however is the confirmation prompt when you remove yourself from the ACL or remove your management rights. We had this before and I don't think it's completely obvious for a user that you, as the event creator, do not have any access or management rights just because you created the event.